File: identity-service/models/Outbox.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const Outbox = sequelize.define('Outbox', {
  id: {
    type: DataTypes.BIGINT,
    primaryKey: true,
    autoIncrement: true
  },
  eventId: {
    type: DataTypes.STRING(36),
    allowNull: false,
    field: 'event_id'
  },
  eventType: {
    type: DataTypes.STRING(255),
    allowNull: false,
    field: 'event_type'
  },
  aggregateId: {
    type: DataTypes.STRING(36),
    allowNull: false,
    field: 'aggregate_id'
  },
  payload: {
    type: DataTypes.TEXT,
    allowNull: true
  },
  status: {
    type: DataTypes.STRING(50),
    allowNull: false,
    defaultValue: 'PENDING'
  },
  createdAt: {
    type: DataTypes.DATE,
    allowNull: false,
    defaultValue: DataTypes.NOW,
    field: 'created_at'
  }
}, {
  tableName: 'outbox',
  timestamps: false
});

module.exports = Outbox;

File: booking-service/src/main/java/lk/ctse/booking/entity/Outbox.java
package lk.ctse.booking.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Entity
@Data
@Table(name = "outbox")
public class Outbox {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true, length = 36)
    private String eventId;

    @Column(nullable = false)
    private String eventType;

    @Column(nullable = false, length = 36)
    private String aggregateId;

    @Column(columnDefinition = "TEXT")
    private String payload;

    @Column(nullable = false, length = 50)
    private String status = "PENDING";

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt = LocalDateTime.now();
}
